From bea489ca3fe47e96c346bd06d3926af0efe80043 Mon Sep 17 00:00:00 2001
From: Jiri Popelka <jpopelka@redhat.com>
Date: Wed, 23 Dec 2015 13:12:36 +0100
Subject: [PATCH] Use unsigned long for dhcp_identifier_length

fixes:
mysql_host_data_source.cc:236:33: error: invalid conversion from 'size_t* {aka unsigned int*}' to 'long unsigned int*' [-fpermissive]
                 bind_[1].length = &dhcp_identifier_length_;
                                 ^
mysql_host_data_source.cc:243:33: error: invalid conversion from 'size_t* {aka unsigned int*}' to 'long unsigned int*' [-fpermissive]
                 bind_[1].length = &dhcp_identifier_length_;
                                 ^
mysql_host_data_source.cc:344:25: error: invalid conversion from 'size_t* {aka unsigned int*}' to 'long unsigned int*' [-fpermissive]
         bind_[1].length = &dhcp_identifier_length_;
---
 src/lib/dhcpsrv/mysql_host_data_source.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/dhcpsrv/mysql_host_data_source.cc b/src/lib/dhcpsrv/mysql_host_data_source.cc
index e5aeb71..bb46a8b 100644
--- a/src/lib/dhcpsrv/mysql_host_data_source.cc
+++ b/src/lib/dhcpsrv/mysql_host_data_source.cc
@@ -515,7 +515,7 @@ private:
     std::vector<uint8_t> dhcp_identifier_;      /// HW address (0) / DUID (1)
     uint8_t     dhcp_identifier_buffer_[DUID::MAX_DUID_LEN];
                                                 /// Buffer for dhcp identifier
-    size_t      dhcp_identifier_length_;        /// Length of dhcp identifier
+    unsigned long dhcp_identifier_length_;      /// Length of dhcp identifier
     uint8_t     dhcp_identifier_type_;		/// Type of dhcp_identifier
     uint32_t    dhcp4_subnet_id_;		/// Subnet DHCPv4 identifier
     uint32_t    dhcp6_subnet_id_;		/// Subnet DHCPv6 identifier
-- 
2.5.0

